src/NuGetGallery.Services/Authentication/ApiKeyV4.cs

```
@@ -135,6 +135,10 @@ private bool TryParseInternal(string plaintextApiKey)
 
                 return success;
             }
+            catch (ArgumentException)
```

Since `ArgumentException` is quite generic, should the `try..catch` be more narrowly targeted at the specific function call that throws when the string is not valid Base32 sequence?

---

> Since `ArgumentException` is quite generic, should `try..catch` be more narrowly targeted at the specific function call that throws when the string is not valid Base32 sequence?

Done!

> On the other hand, should any exception thrown cause the function to return false?

At this point, all known error conditions are handled. Given API keys are crucial to nuget.org's authentication, I would rather fail fast if there are any unexpected error conditions. Swallowing exceptions may result in us incorrectly handling malformed API keys - this would be really hard to catch after the fact.